### ISSUE TITLE
calico-upgrade tool flags preDNAT policies with egress type

### ISSUE THREAD

**Issue by bradbehle** [projectcalico/calico#1796]
When running the calico-upgrade tool, it flags any preDNAT policies that have an egress type and won't migrate the calico data.  Since these policies function fine in calico v2, it would be good for this tool to migrate these policies into a v3 policy that matches the function of the v2 policy.

## Expected Behavior
calico-upgrade should migrate this policy successfully, probably by just removing the "egress" type since it isn't being honored in calico v2 anyway, and since that is what the manual step the customer will need to take is.

## Current Behavior
calico-upgrade currently errors out of the calico-upgrade call (or reports the error in the calico-upgrade dry-run call)

## Possible Solution
calico-upgrade should migrate this policy successfully, probably by just removing the "egress" type since it isn't being honored in calico v2 anyway, and since that is what the manual step the customer will need to take is.

## Steps to Reproduce (for bugs)
1. Create the following policy or something similar and apply it to a calico v2 cluster (2.6.5 or higher to allow calico-upgrade to be run):

~~~
- apiVersion: v1
  kind: policy
  metadata:
    name: custom-allow
  spec:
    ingress:
    - action: allow
      destination: {}
      source:
        nets:
        - 10.10.10.2/32
        - 10.10.10.3/32
        - 10.10.10.4/32
    order: 200
    preDNAT: true
    selector: role in { 'worker', 'master' }
    types:
    - ingress
    - egress
~~~

2. Run calico-upgrade against that cluster specifying dry-run (or start I guess), and it will flag that policy with an error like `v1 name: Policy(name=custom-allow)v3 name: GlobalNetworkPolicy(custom-allow)error: error with field PolicySpec.Types = '[Ingress Egress]' (PreDNAT PolicySpec cannot have 'egress' Type)v3 resource:

## Context
We have many customer clusters to upgrade in an automated way, and manually updating clusters that have this issue is problematic.

## Your Environment
* Calico version: 2.6.5  (upgrading to 3.0.3)
* Orchestrator version (e.g. kubernetes, mesos, rkt): kube 1.9 (upgrading to kube 1.10)
* Operating System and version: Ubuntu 16.04 LTS
* Link to your project (optional):

**Comment by tmjd** [comment]
@bradbehle I'm working on a fix for this but I've found that I can't load the yaml you included with calicoctl from v2.6.5.  What I found was that I could load something similar (without the types) with calicoctl from v2.5.1 and then when that was queried with calicoctl from v2.6.5 it would dump yaml like you included.
With the changes I have made and the policy I've mentioned above I can run calico-upgrade successfully where before it would produce the error you were seeing.

I just wanted to mention the above to you to ensure that you think this sounds reasonable and that it will fix the issue.  I believe it will but since I had to set up things slightly different from what you suggested I wanted to run it by you.

**Comment by bradbehle** [comment]
@tmjd Yep, this sounds like what I'm asking for.  Most likely these policies that I was having problems with were created with an earlier version of calico, so that matches our scenario.  If calico-upgrade was failing before but succeeding now on policies like this, that's what we want.  I would be happy to test out a new calico-upgrade binary when you have one available.

**Comment by tmjd** [comment]
The v1.0.3 release of calico-upgrade https://github.com/projectcalico/calico-upgrade/releases/tag/v1.0.3 has the fixes for this issue.
